src/Azure.Functions.ArtifactAssembler/CliArtifactZipper.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO.Compression;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Azure.Functions.ArtifactAssembler
+{
+    internal class CliArtifactZipper
+    {
+        private readonly string _rootWorkingDirectory;
+        public CliArtifactZipper(string rootWorkingDirectory)
+        {
+            _rootWorkingDirectory = rootWorkingDirectory;
+
+        }
+        internal void ZipCliArtifacts()
+        {
+            string stagingDirectory = Path.Combine(_rootWorkingDirectory, Constants.StagingDirName, Constants.CliOutputArtifactDirectoryName);
+
+            // Get all directories in the staging directory
+            var directories = Directory.GetDirectories(stagingDirectory);
+
+            foreach (var dir in directories)
+            {
+                string workersPath = Path.Combine(dir, "workers");
+
+                // Ensure 'workers' directory exists
+                if (!Directory.Exists(workersPath))
+                {
+                    Directory.CreateDirectory(workersPath);
+                    Console.WriteLine($"Created missing 'workers' directory in {dir}");
+                }
+                else
+                {
+                    Console.WriteLine($"'workers' directory already exists in {dir}");
+                }
+
+                // Add placeholder file if 'workers' directory is empty
```

Comment:
We had an issue that the `workers` directory was present before when creating the CLI artifacts but the powershell Compress-Archive function would delete the workers directory if it was empty. I just wanted to add this here as a safety check since this is the step that would run right as we  zip.

I'll extract this logic to a smaller method then :)